<!-- Spawns minimum snap trajectory planner and sampler -->

<launch>
    <arg name="drone_id" default="0"/>

    <!-- Input/output topics -->
    <arg name="odometry_topic" default="mavros/local_position/odom" />

    <!--- Trajectory planner -->
    <node name="trajectory_planner" pkg="trajectory_planner" type="trajectory_generator" output="screen">
        <param name="/max_v" value="1" type="double"/> <!-- 1.5 m/s -->
        <param name="/max_a" value="3" type="double"/> <!-- 6.0 m/s^2 -->
        <!-- subscribed topics -->
        <remap from="uav_pose" to="/$(arg odometry_topic)"/>
        <remap from="/waypoints" to="/planner/goals"/>
        <!-- published topics -->
        <remap from="trajectory" to="/planner/sample_trajectory"/>
    </node>

    <!--- Trajectory sampler -->
    <node name="trajectory_sampler" pkg="trajectory_planner" type="trajectory_sampler" output="screen">
        <!-- subscribed topics -->
        <remap from="path_segments_4D" to="/planner/sample_trajectory"/>
        <!-- published topics -->
        <remap from="command/trajectory" to="/planner/trajectory"/>
    </node>

</launch>
